cypress/integration/basic/site.spec.js

```diff
@@ -0,0 +1,39 @@
+describe('Homepage and login', () => {
+
+  beforeEach(() => {
+     cy.visit('http://localhost:3000')
+  })
+
+it('Should go to /welcome and return HTTP 200',() => {
+  cy.request({
+    url: 'http://localhost:3000/welcome',
+    followRedirect: false  // turns off redirects
+  })
+  .then((response) => {
+     cy.expect(response.status).to.eq(200)
+  })
+})
+
+it('Test that /applications redirects to CAS ',() => {
+  const username = Cypress.env('userName')
+  const password = Cypress.env('password')
+
+  cy.request({
```

Comment:
When an unauthenticated user goes straight to "/applications", they should be automatically redirected to CAS. Change this test instead so that it loads /applications directly and confirms that a redirect to CAS was received. Also confirm (if possible) before this that the user is not already authenticated (probably available by checking for a cookie).